File: backend/app/domain/documents.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID


@dataclass(frozen=True)
class Document:
    id: UUID
    project_id: UUID
    display_name: str
    original_filename: str
    file_type: str
    size_bytes: int
    content_hash: str
    status: str
    stage: str
    error_code: str | None
    error_message: str | None
    page_count: int | None
    original_name: str
    created_at: datetime
    updated_at: datetime


class DocumentError(Exception):
    def __init__(self, code: str, message: str, status_code: int = 400):
        self.code = code
        self.message = message
        self.status_code = status_code
        super().__init__(message)

File: frontend/src/types/documents.ts
export interface DocumentItem {
  id: string;
  project_id: string;
  display_name: string;
  original_filename: string;
  file_type: "pdf" | "docx" | "txt" | "md";
  size_bytes: number;
  content_hash: string;
  status: "processing" | "ready" | "failed";
  stage: string;
  error_code: string | null;
  error_message: string | null;
  page_count: number | null;
  created_at: string;
  updated_at: string;
}

export interface Segment {
  index: number;
  kind: string;
  label: string;
  page_number: number | null;
  paragraph_number: number | null;
  text: string;
}

export interface ImportOutcome {
  filename: string;
  document: DocumentItem | null;
  error_code: string | null;
  error_message: string | null;
}
